From 857fba02db1688fc8891723bc639b2050171c769 Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Mon, 26 Nov 2018 10:44:30 +0800
Subject: [PATCH] arch: arm64: instead of enable_daif with enable_da_f

When enable below lock detect configure,
CONFIG_DEBUG_RT_MUTEXES=y
CONFIG_DEBUG_SPINLOCK=y
CONFIG_DEBUG_MUTEXES=y
CONFIG_DEBUG_WW_MUTEX_SLOWPATH=y
CONFIG_DEBUG_LOCK_ALLOC=y
CONFIG_PROVE_LOCKING=y
There will be a warning trace as below:
DEBUG_LOCKS_WARN_ON(!irqs_disabled())
------------[ cut here ]------------
WARNING: CPU: 0 PID: 96 at /kernel-source/kernel/locking/lockdep.c:
2779 trace_hardirqs_off_caller+0x88/0x90

Refer to commit 0f15adbb2861 ("arm64: Add skeleton to harden the
branch predictor against aliasing attacks") and commit 5dfc6ed27710
("arm64: entry: Apply BP hardening for high-priority synchronous exceptions"),
Change function enable_daif into enable_da_f, other modification of
these 2 commits had been merged in advance.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/kernel/entry.S |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/kernel/entry.S b/arch/arm64/kernel/entry.S
index b84c9de..60a84c5 100644
--- a/arch/arm64/kernel/entry.S
+++ b/arch/arm64/kernel/entry.S
@@ -749,7 +749,7 @@ el0_ia:
 	 * Instruction abort handling
 	 */
 	mrs	x26, far_el1
-	enable_daif
+	enable_da_f
 #ifdef CONFIG_TRACE_IRQFLAGS
 	bl	trace_hardirqs_off
 #endif
@@ -784,7 +784,7 @@ el0_sp_pc:
 	 * Stack or PC alignment exception handling
 	 */
 	mrs	x26, far_el1
-	enable_daif
+	enable_da_f
 #ifdef CONFIG_TRACE_IRQFLAGS
 	bl	trace_hardirqs_off
 #endif
-- 
1.7.9.5

